src/backend/cdb/cdbtm.c

```diff
@@ -1608,6 +1608,14 @@ doQEDistributedExplicitBegin()
 static bool
 isDtxQueryDispatcher(void)
 {
+#ifdef SERVERLESS
```

Comment:
Add a guc to control?
For example:

    return (Gp_role == GP_ROLE_DISPATCH &&
            isDtmStarted &&
            isSharedLocalSnapshotSlotPresent);

Reply:
add global variable enable_serverless, default to false, set to true in plugin.